packages/clay-progress-bar/src/ClayProgressBar.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import warning from 'warning';
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Flag to indicate if `children` should be wrapped as `feedback`.
+	 */
+	feedback?: boolean;
+
+	/**
+	 * The current value of the progress bar. Should range from 0 to 100.
+	 */
+	value: number;
+
+	/**
+	 * Flag to indicate whether a "warning" color for the bar.
+	 */
+	warn?: boolean;
+}
+
+const ClayProgressBar: React.FunctionComponent<Props> = ({
+	children,
+	className,
+	feedback = true,
+	value = 0,
+	warn,
+	...otherProps
+}) => {
+	warning(
+		value >= 0 || value <= 100,
+		'ClayProgressBar requires `value` to be in the range of 0 to 100'
+	);
```

Comment:
Instead of just warning, we should probably actually clamp the value, otherwise we would end up rendering with `width: 1000%` and such.

```typescript
export default function clamp(
  value: number,
  minimum: number,
  maximum: number,
): number {
  if (minimum <= maximum) {
    return Math.min(Math.max(value, minimum), maximum);
  } else {
    return NaN;
  }
}
```